src/Clock/Clock.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+type Props = {
+  name: string,
+};
+
+type State = {
+  today: Date,
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    today: new Date(),
+  };
+
+  newTimer = 0;
+
+  componentDidMount() {
+    this.newTimer = window.setInterval(() => {
+      this.setState({ today: new Date() });
+      /* eslint-disable no-console */
+      console.info(this.state.today.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { name } = this.props;
+
+    if (prevProps.name !== name) {
+      console.debug(`Renamed from ${prevProps.name} to ${name}`);
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.newTimer);
+  }
+
+  render() {
```

Comment:
The Clock should be visible only when a certain condition is met. Use a prop or state to set this condition and render the Clock component accordingly.

Reply:
but there is  {isClockVisible && <Clock name={clockName} />} and it work... in my oppinion